src/common/translations/localize.ts

```diff
@@ -116,17 +94,33 @@ export const computeLocalize = async <Keys extends string = LocalizeKeys>(
       } catch (err: any) {
         return "Translation error: " + err.message;
       }
-      cache._localizationCache[messageKey] = translatedMessage;
+      localizationCache[translatedValue] = translatedMessage;
     }
 
     let argObject = {};
+    // TS says that we will always get the new format, but we might get the old format
+    const args = _args as any;
     if (args.length === 1 && typeof args[0] === "object") {
       argObject = args[0];
-    } else {
+    } else if (args.length >= 2) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        "[FIXME] While localizing",
+        key,
+        "the old format for parameters was used. See https://developers.home-assistant.io/blog/2023/09/27/localize-handling for more info."
+      );
       for (let i = 0; i < args.length; i += 2) {
         argObject[args[i]] = args[i + 1];
       }
     }
+    if (Object.values(argObject).some((v) => v === undefined)) {
+      // eslint-disable-next-line no-console
+      console.warn(
```

Comment:
I think this should be logged as an error instead.

@bramkragten  should we throw here in development?

Reply:
Don't think we should throw here (even in development) because:
- I would rather `hass.localize` work exactly as before
- There are some cases where undefined could be passed as we just have a non-null assertion
- If it threw, it would break the page

Reply:
It should be `console.error` regardless.

> There are some cases where undefined could be passed as we just have a non-null assertion

Where? Seems like those cases should be fixed.

Reply:
> Seems like those cases should be fixed.

To clarify, I'm referring to cases where Typescript has no way of knowing that it's not going to be null, but the existing code trusts that it's not going to be null.

Reply:
😕 There are a lot of instances touched here.  Can you point me to which ones you are referring?

Reply:
Sure, here are some places where a non-null assertion had to be added:
- `hui-energy-gas-graph-card`
- `hui-energy-water-graph-card`
- `ha-config-repairs`